src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+    private final int sideOfFigureA = random.nextInt(1,10);
+    private final int sideOfFigureB = random.nextInt(1,10);
+    private final int height = random.nextInt(1, 10);
+
+    public Figure getRandomFigure() {
+        Figure[] figure = new Figure[] {new Square("square",
+                sideOfFigureA, colorSupplier.getRandomColor()),
+                new RightTriangle("RightTriangle", sideOfFigureA, sideOfFigureB,
+                colorSupplier.getRandomColor()),
+                new Circle("circle", sideOfFigureA,
+                colorSupplier.getRandomColor()),
+                new Rectangle("rectangle", sideOfFigureA, sideOfFigureB,
+                colorSupplier.getRandomColor()),
+                new IsoscelesTrapezoid("IsoscelesTrapezoid", sideOfFigureA, sideOfFigureB,
+                height, colorSupplier.getRandomColor())};
```

Comment:
You create several objects to get only 1 of them. It's not good.
Please add logic to determine random figure type and then create only 1 object